tests/unit/cluster/hostnames.tcl

```diff
@@ -170,10 +170,7 @@ test "Verify the nodes configured with prefer hostname only show hostname for ne
     } else {
         fail "Node did not learn about the 2 shards it can talk to"
     }
-    set slot_result [R 6 CLUSTER SLOTS]
-    assert_equal [lindex [get_slot_field $slot_result 0 2 3] 1] "shard-1.com"
-    assert_equal [lindex [get_slot_field $slot_result 1 2 3] 1] "shard-2.com"
-    assert_equal [lindex [get_slot_field $slot_result 2 2 3] 1] "shard-3.com"
+    wait_for_cluster_propagation
```

Comment:
I feel that it would be better to keep the original L173-L176 and insert a `wait_for_cluster_propagation` call right before the checks.

Also, I think there is similar flakiness on L153-L155.

Reply:
I've applied a fix around L153-155, WDYT about it, I'm thinking of applying it similarly on L173-L176 to avoid flakiness.